saplings/circuits/src/components/Content.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2018-2020 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useReducer } from 'react';
+import { useLocalNodeState } from '../state/localNode';
+import mockCircuits from '../mockData/mockCircuits';
+import mockProposals from '../mockData/mockProposals';
+import { processCircuits } from '../data/processCircuits';
+
+import './Content.scss';
+
+const circuitsReducer = (state, action) => {
+  switch (action.type) {
+    case 'sort': {
+      const sortedCircuits = action.sortCircuits(
+        state.filteredCircuits,
+        action.sort
+      );
+      return { ...state, filteredCircuits: sortedCircuits };
+    }
+    case 'filter': {
+      const filteredCircuits = action.filterCircuits(
+        state.circuits,
+        action.filter
+      );
+      return { ...state, filteredCircuits };
+    }
+    default:
+      throw new Error(`unhandled action type: ${action.type}`);
+  }
+};
+const Content = () => {
+  const circuits = processCircuits(mockCircuits.concat(mockProposals));
+
+  const [circuitState, circuitsDispatch] = useReducer(circuitsReducer, {
+    circuits,
+    filteredCircuits: circuits
+  });
+  const nodeID = useLocalNodeState();
+  const totalCircuits = circuitState.circuits.length;
+  const actionRequired = circuitState.circuits.filter(circuit =>
+    circuit.actionRequired(nodeID)
+  ).length;
+
+  return (
+    <div className="content">
```

Comment:
With a pretty generic class like `content`, I would suggest adding an ID to the top element to prevent collisions with styles in other files/elements

Reply:
Or, if there can be more than one on a page at a time, then a more specific classname

Reply:
I updated the name of the class. That's a good tip!